cg/store/filters/status_case_sample_filters.py

```diff
@@ -18,12 +18,51 @@ def filter_cases_with_sample_by_internal_id(case_samples: Query, sample_internal
     return case_samples.filter(Sample.internal_id == sample_internal_id)
 
 
+def filter_by_order(case_samples: Query, order_id: int, **kwargs) -> Query:
+    return case_samples.join(Order, Case.orders).filter(Order.id == order_id)
```

Comment:
Is there any point in adding the .distinct here? Or I guess we will never use it without the subsequent filters which already do the .distinct

Reply:
Added `distinct`